server/src/main/java/org/opensearch/env/NodeEnvironment.java

```diff
@@ -298,22 +299,32 @@ public void close() {
     }
 
     public NodeEnvironment(Settings settings, Environment environment) throws IOException {
-        this(settings, environment, IndexStoreListener.EMPTY);
+        this(settings, environment, org.opensearch.index.store.IndexStoreListener.EMPTY);
+    }
+
+    /**
+     * Use {@link NodeEnvironment#NodeEnvironment(Settings, Environment, org.opensearch.index.store.IndexStoreListener)} instead.
+     * This constructor is kept around on 2.x to avoid breaking changes.
+     */
+    @Deprecated(forRemoval = true, since = "2.19.0")
+    public NodeEnvironment(Settings settings, Environment environment, IndexStoreListener indexStoreListener) throws IOException {
+        this(settings, environment);
```

Comment:
We could also throw `UnsupportedOperationException` here instead as well

Reply:
```
this(settings, environment, (org.opensearch.index.store.IndexStoreListener) indexStoreListener);
```